--- a/drivers/power/supply/power_supply_leds.c	2016-04-12 11:09:26.000000000 -0500
+++ b/drivers/power/supply/power_supply_leds.c	2017-03-01 21:53:03.740678198 -0600
@@ -35,6 +35,10 @@
 		led_trigger_event(psy->charging_full_trig, LED_FULL);
 		led_trigger_event(psy->charging_trig, LED_OFF);
 		led_trigger_event(psy->full_trig, LED_FULL);
+		/* For charge blink / full solid, turn off the led first
+		 * then turn it on full brightness */
+		led_trigger_event(psy->charging_blink_full_solid_trig,
+			LED_OFF);
 		led_trigger_event(psy->charging_blink_full_solid_trig,
 			LED_FULL);
 		break;
